plugins/woocommerce-admin/client/payments-welcome/index.tsx

```diff
@@ -53,87 +42,44 @@ interface activatePromoResponse {
 	success: boolean;
 }
 
-const PROMO_NAME = 'wcpay-promo-2022-us-incentive-20-off';
-
-const LearnMore = () => {
-	const handleClick = () => {
-		recordEvent( 'wcpay_welcome_learn_more', {} );
-	};
-	return (
-		<a
-			onClick={ handleClick }
-			href="https://woocommerce.com/payments/"
-			target="_blank"
-			rel="noreferrer"
-		>
-			{ strings.learnMore }
-		</a>
-	);
-};
-
-export const PaymentMethods = () => (
-	<div className="wcpay-connect-account-page-payment-methods">
-		<Visa />
-		<MasterCard />
-		<Maestro />
-		<Amex />
-		<DinersClub />
-		<CB />
-		<Discover />
-		<UnionPay />
-		<JCB />
-		<ApplePay />
-	</div>
-);
-
-const TermsOfService = () => (
-	<p className="wcpay-connect-account-page-terms-of-service">
-		{ strings.terms }
-	</p>
-);
-
-const ConnectPageError = ( { errorMessage }: { errorMessage: string } ) => {
-	if ( ! errorMessage ) {
-		return null;
-	}
-	return (
-		<Notice
-			className="wcpay-connect-error-notice"
-			status="error"
-			isDismissible={ false }
-		>
-			{ errorMessage }
-		</Notice>
-	);
-};
-
-const ConnectPageOnboarding = ( {
-	isJetpackConnected,
-	installAndActivatePlugins,
-	setErrorMessage,
-	connectUrl,
-	enabledApms,
-}: {
-	isJetpackConnected?: boolean;
-	installAndActivatePlugins: PluginsStoreActions[ 'installAndActivatePlugins' ];
-	// eslint-disable-next-line @typescript-eslint/ban-types
-	setErrorMessage: Function;
-	connectUrl: string;
-	enabledApms: Set< Apm >;
-} ) => {
+const ConnectAccountPage = () => {
+	const incentive = getAdminSetting( 'wcpayWelcomePageIncentive' );
+	const { updateOptions } = useDispatch( OPTIONS_STORE_NAME );
+	const { installAndActivatePlugins } = useDispatch( 'wc/admin/plugins' );
 	const [ isSubmitted, setSubmitted ] = useState( false );
-	const [ isNoThanksClicked, setNoThanksClicked ] = useState( false );
+	const [ errorMessage, setErrorMessage ] = useState( '' );
+	const [ enabledApms, setEnabledApms ] = useState( new Set< Apm >() );
 
-	const [ isExitSurveyModalOpen, setExitSurveyModalOpen ] = useState( false );
+	const { isJetpackConnected, connectUrl } = useSelect( ( select ) => {
+		return {
+			isJetpackConnected:
+				select( 'wc/admin/plugins' ).isJetpackConnected(),
+			connectUrl:
+				'admin.php?wcpay-connect=1&_wpnonce=' +
+				getAdminSetting( 'wcpay_welcome_page_connect_nonce' ),
+		};
+	} );
 
-	const renderErrorMessage = ( message: string ) => {
-		setErrorMessage( message );
-		setSubmitted( false );
-	};
+	/**
+	 * Record page view and save viewed timestamp.
+	 */
+	useEffect( () => {
+		recordEvent( 'page_view', {
+			path: 'payments_connect_core_test',
+			incentive_id: incentive.id,
+		} );
+		updateOptions( {
+			wcpay_welcome_page_viewed_timestamp: Math.floor(
```

Comment:
Yes, it was only updated for the first time before, but it doesn't work as intended. Because we actually want to know that was the last time they viewed it, to send a follow-up notice 🙁